woocommerce/woocommerce-admin
# Add images to homepage template and a new layout for sites with more than 4 products

Closes #3078, Closes #2902.

This PR adds another template for sites with more than 4 products. It borrows logic from the default storefront template.

This PR also uses CC0 images to set some example cover and media block images based on industry.

### Screenshots

#### Template for sites with less than 4 products

<img width="698" alt="Screen Shot 2019-10-30 at 2 51 56 PM" src="https://user-images.githubusercontent.com/689165/67889270-e4f27480-fb24-11e9-9a2b-747917588732.png">

<img width="672" alt="Screen Shot 2019-10-30 at 2 31 18 PM" src="https://user-images.githubusercontent.com/689165/67888606-cfc91600-fb23-11e9-96fa-ab1b54aa2080.png">

#### Template for sites with more than 4 products

<img width="634" alt="Screen Shot 2019-10-30 at 2 32 40 PM" src="https://user-images.githubusercontent.com/689165/67888638-dc4d6e80-fb23-11e9-9d36-b3b545710a98.png">

#### Uninitialized media block if uploading fails

<img width="1127" alt="Screen Shot 2019-10-29 at 4 03 55 PM" src="https://user-images.githubusercontent.com/689165/67889121-a5c42380-fb24-11e9-9ac1-f425a71f69f8.png">

### Detailed test instructions:

* Delete your homepage.
* Make sure you have less than 4 products.
* Walk through the profile wizard and select at least one industry (if you selected multiple industries, images will be pulled from each).
* Finish the profile wizard.
* Go to the appearance task, and create a homepage.
* A homepage template should be created. A cover block containing one of the industry images should be present. There should also be a few media text blocks using industry images.
* Delete your homepage.
* Make sure you have more than 4 products.
* Go to the appearance task, and create a homepage.
* A homepage template should be created. A cover block containing one of the industry images should be present.


* Optional: Replace the `$images = self::sideload_homepage_images....` calls with `$images = array();` and test what happens if the sideloading fails (you should see a cover block with a background color instead -- and uninitialized media blocks).
